### PR TITLE
Reuse `MacHeader`

### PR DESCRIPTION
**Pull request by bastbu**
# PR for issue #969

## What is being addressed

Reuse `MacHeader` when possible.

### CODE REVIEW

**Review by MaggieSalak — APPROVED** [review]
Looks good, I reviewed it and didn't see anything needing improvement.

File: Tests/Unit/NetworkServer/LnsProtocolMessageProcessorTests.cs [hunk 1]
```diff
@@ -332,7 +332,7 @@ public async Task InternalHandleDataAsync_ShouldProperlyCreateLoraPayloadForJoin
                                                 'DevNonce':54360,'MIC':-1056607131,'RefTime':0.000000,'DR':5,'Freq':868300000,'upinfo':{'rctx':0,
                                                 'xtime':68116944405337035,'gpstime':0,'fts':-1,'rssi':-53,'snr':8.25,'rxtime':1636131701.731686}}");
             var expectedRadioMetadata = GetExpectedRadioMetadata();
-            var expectedMhdr = new byte[] { 0 };
+            var expectedMhdr = new MacHeader(0);
```
Comment (atifaziz) [inline Tests/Unit/NetworkServer/LnsProtocolMessageProcessorTests.cs:335]: I think we should use the more meaningful constructor here:

```suggestion
            var expectedMhdr = new MacHeader(MacMessageType.JoinRequest);
```

File: Tests/Unit/NetworkServer/LnsProtocolMessageProcessorTests.cs [hunk 2]
```diff
@@ -297,7 +297,7 @@ public async Task InternalHandleDataAsync_ShouldProperlyCreateLoraPayloadForUpdf
                                                 'MIC':45234788,'RefTime':0.000000,'DR':5,'Freq':868300000,'upinfo':{'rctx':0,'xtime':68116944405337035,
                                                 'gpstime':0,'fts':-1,'rssi':-53,'snr':8.25,'rxtime':1636131701.731686}}");
             var expectedRadioMetadata = GetExpectedRadioMetadata();
-            var expectedMhdr = new byte[] { 128 };
+            var expectedMhdr = new MacHeader(128);
```
Comment (atifaziz) [inline Tests/Unit/NetworkServer/LnsProtocolMessageProcessorTests.cs:300]: I think we should use the more meaningful constructor here:

```suggestion
            var expectedMhdr = new MacHeader(MacMessageType.ConfirmedDataUp);
```

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs [hunk 3]
```diff
@@ -42,22 +42,10 @@ public LoRaPayloadJoinRequest()
...
-            DevEUI = new Memory<byte>(inputMessage, 9, 8);
-            // get the DevNonce
-            DevNonce = DevNonce.Read(inputMessage.AsSpan(17));
-        }
-
         public LoRaPayloadJoinRequest(string appEUI, string devEUI, DevNonce devNonce)
         {
             // Mhdr is always 0 in case of a join request
-            Mhdr = new byte[1] { 0x00 };
+            MHdr = new MacHeader(0x00);
```
Comment (atifaziz) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinRequest.cs:48]: I think we should use the more meaningful constructor here:

```suggestion
            MHdr = new MacHeader(MacMessageType.JoinRequest);
```

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinAccept.cs [hunk 4]
```diff
@@ -195,8 +195,10 @@ public override byte[] PerformEncryption(AppKey key)
 
         public override byte[] GetByteMessage()
```
Comment (atifaziz) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinAccept.cs:196]: nit: could turn this into a one-liner, skipping the list:

```suggestion
        public override byte[] GetByteMessage() => RawMessage.Prepend((byte)MHdr).ToArray();
```

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinAccept.cs [hunk 5]
```diff
@@ -67,8 +67,8 @@ public LoRaPayloadJoinAccept(string netId, DevAddr devAddr, byte[] appNonce, byt
 
             var cfListLength = cfList == null ? 0 : cfList.Length;
             RawMessage = new byte[1 + 12 + cfListLength];
-            Mhdr = new Memory<byte>(RawMessage, 0, 1);
-            Array.Copy(new byte[] { 32 }, 0, RawMessage, 0, 1);
+            MHdr = new MacHeader(32);
```
Comment (atifaziz) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinAccept.cs:70]: I think we should use the more meaningful constructor here:

```suggestion
            MHdr = new MacHeader(MacMessageType.JoinAccept);
```

**Review by atifaziz — APPROVED** [review]
A couple of suggestions otherwise LGTM.

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinAccept.cs (hunk 4 above)
Comment (atifaziz) [reply LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinAccept.cs]: Sorry, but this should have never been a _suggested change that could be applied_ immediately. On GitHub, I couldn't somehow mark the suggestion as a replacement for the whole function, but then forgot to revert the suggestion into a regular code snippet before posting my comment. In short, [this has broken the build](https://github.com/Azure/iotedge-lorawan-starterkit/runs/4760665538?check_suite_focus=true#step:8:36).

**Pull request merged**